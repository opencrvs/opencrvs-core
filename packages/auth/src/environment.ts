/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * OpenCRVS is also distributed under the terms of the Civil Registration
 * & Healthcare Disclaimer located at http://opencrvs.org/license.
 *
 * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
 */

import { cleanEnv, str, port, url, num, bool } from 'envalid'

export const env = cleanEnv(process.env, {
  REDIS_HOST: str({ devDefault: 'localhost' }),
  AUTH_HOST: str({ default: '0.0.0.0' }),
  AUTH_PORT: port({ default: 4040 }),
  USER_MANAGEMENT_URL: url({ devDefault: 'http://localhost:3030/' }),
  METRICS_URL: url({ devDefault: 'http://localhost:1050' }),
  NOTIFICATION_SERVICE_URL: url({ devDefault: 'http://localhost:2020/' }),
  DOMAIN: str({ devDefault: '*' }),
  COUNTRY_CONFIG_URL: url({ devDefault: 'http://localhost:3040/' }), // used for external requests (CORS whitelist)
  COUNTRY_CONFIG_URL_INTERNAL: url({ devDefault: 'http://localhost:3040/' }), // used for internal service-to-service communication
  LOGIN_URL: url({ devDefault: 'http://localhost:3020/' }),
  CLIENT_APP_URL: url({ devDefault: 'http://localhost:3000/' }),
  CERT_PRIVATE_KEY_PATH: str({ devDefault: '../../.secrets/private-key.pem' }),
  CERT_PUBLIC_KEY_PATH: str({ devDefault: '../../.secrets/public-key.pem' }),
  SENTRY_DSN: str({ default: undefined }),
  QA_ENV: bool({ default: false }),

  CONFIG_TOKEN_EXPIRY_SECONDS: num({ default: 604800 }), // 1 week
  CONFIG_SMS_CODE_EXPIRY_SECONDS: num({ default: 600 }), // 10 minutes
  CONFIG_SYSTEM_TOKEN_EXPIRY_SECONDS: num({ default: 600 }) // 10 minutes
})
