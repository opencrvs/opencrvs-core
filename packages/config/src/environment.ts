/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * OpenCRVS is also distributed under the terms of the Civil Registration
 * & Healthcare Disclaimer located at http://opencrvs.org/license.
 *
 * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
 */
import { cleanEnv, str, port, url, bool } from 'envalid'

export const env = cleanEnv(process.env, {
  HOST: str({ devDefault: 'localhost' }),
  DOMAIN: str({ devDefault: '*' }),
  LOGIN_URL: url({ devDefault: 'http://localhost:3020/' }),
  CLIENT_APP_URL: url({ devDefault: 'http://localhost:3000/' }),
  PORT: port({ default: 2021 }),
  QA_ENV: bool({ default: false }),
  CHECK_INVALID_TOKEN: bool({
    devDefault: false,
    desc: `Check if the token has been invalided in the auth service before it has expired`
  }),
  CERT_PUBLIC_KEY_PATH: str({ devDefault: '../../.secrets/public-key.pem' }),
  SENTRY_DSN: str({ default: undefined }),
  AUTH_URL: url({ devDefault: 'http://localhost:4040/' }),
  COUNTRY_CONFIG_URL: url({ devDefault: 'http://localhost:3040/' }),
  MONGO_URL: url({ devDefault: 'mongodb://localhost/application-config' }),
  USER_MANAGEMENT_URL: url({ devDefault: 'http://localhost:3030/' }),
  FHIR_URL: url({ devDefault: 'http://localhost:3447/fhir' }),
  HEARTH_MONGO_URL: str({ default: 'mongodb://localhost/hearth-dev' })
})
