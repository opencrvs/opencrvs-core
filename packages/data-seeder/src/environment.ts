/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * OpenCRVS is also distributed under the terms of the Civil Registration
 * & Healthcare Disclaimer located at http://opencrvs.org/license.
 *
 * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
 */
import { bool, cleanEnv, str, url } from 'envalid'

export const env = cleanEnv(process.env, {
  AUTH_HOST: url({ devDefault: 'http://localhost:4040' }),
  COUNTRY_CONFIG_HOST: url({ devDefault: 'http://localhost:3040' }),
  GATEWAY_HOST: url({ devDefault: 'http://localhost:7070' }),
  SUPER_USER_PASSWORD: str({ devDefault: 'password' }),
  ACTIVATE_USERS: bool({ devDefault: true })
})
