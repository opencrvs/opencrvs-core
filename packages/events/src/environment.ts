/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * OpenCRVS is also distributed under the terms of the Civil Registration
 * & Healthcare Disclaimer located at http://opencrvs.org/license.
 *
 * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
 */

import { cleanEnv, url } from 'envalid'

/**
 * When defining variables aim to be consistent with existing values.
 * Define URLs without trailing slashes, include the protocol.
 */
export const env = cleanEnv(process.env, {
  EVENTS_MONGO_URL: url({ devDefault: 'mongodb://localhost/events' }),
  USER_MGNT_MONGO_URL: url({ devDefault: 'mongodb://localhost/user-mgnt' }),
  ES_URL: url({ devDefault: 'http://localhost:9200' }),
  COUNTRY_CONFIG_URL: url({ devDefault: 'http://localhost:3040' }),
  DOCUMENTS_URL: url({ devDefault: 'http://localhost:9050' }),
  USER_MANAGEMENT_URL: url({ devDefault: 'http://localhost:3030' }),
  AUTH_URL: url({ devDefault: 'http://localhost:4040' })
})
