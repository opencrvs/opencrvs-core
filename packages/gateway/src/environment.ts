/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * OpenCRVS is also distributed under the terms of the Civil Registration
 * & Healthcare Disclaimer located at http://opencrvs.org/license.
 *
 * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
 */
import { cleanEnv, str, port, url, bool, num } from 'envalid'

export const env = cleanEnv(process.env, {
  REDIS_HOST: str({ devDefault: 'localhost' }),
  HOST: str({ devDefault: '0.0.0.0' }),
  PORT: port({ default: 7070 }),
  DOMAIN: str({ devDefault: '*' }),
  LOGIN_URL: url({ devDefault: 'http://localhost:3020/' }),
  CLIENT_APP_URL: url({ devDefault: 'http://localhost:3000/' }),
  FHIR_URL: url({
    devDefault: 'http://localhost:3447/fhir'
  }),
  CERT_PUBLIC_KEY_PATH: str({
    devDefault: '../../.secrets/public-key.pem'
  }),
  SEARCH_URL: url({ devDefault: 'http://localhost:9090/' }),
  METRICS_URL: url({ devDefault: 'http://localhost:1050' }),
  AUTH_URL: url({ devDefault: 'http://localhost:4040' }),
  USER_MANAGEMENT_URL: url({ devDefault: 'http://localhost:3030/' }),
  WEBHOOKS_URL: url({ devDefault: 'http://localhost:2525/' }),
  APPLICATION_CONFIG_URL: url({ devDefault: 'http://localhost:2021/' }),
  NOTIFICATION_URL: url({ devDefault: 'http://localhost:2020/' }),
  WORKFLOW_URL: url({ devDefault: 'http://localhost:5050/' }),
  EVENTS_URL: url({ devDefault: 'http://localhost:5555/' }),
  COUNTRY_CONFIG_URL: url({ devDefault: 'http://localhost:3040' }),
  DOCUMENTS_URL: url({ devDefault: 'http://localhost:9050' }),
  DISABLE_RATE_LIMIT: bool({
    default: false,
    desc: 'Disables the Redis-based rate limiting globally'
  }),
  SENTRY_DSN: str({ default: undefined }),
  QA_ENV: bool({ default: false }),
  CHECK_INVALID_TOKEN: bool({
    devDefault: false,
    desc: 'Check if the token has been invalidated in the auth service before it has expired'
  }),
  LANGUAGES: str({ devDefault: 'bn,en' }),
  COUNTRY: str({ devDefault: 'FAR' }),
  CONFIG_TOKEN_EXPIRY_SECONDS: num({ default: 604800 }), // 1 week
  CONFIG_SMS_CODE_EXPIRY_SECONDS: num({ default: 600 }), // 10 minutes
  CONFIG_SYSTEM_TOKEN_EXPIRY_SECONDS: num({ default: 600 }), // 10 minutes
  MINIO_BUCKET: str({ devDefault: 'ocrvs' })
})
