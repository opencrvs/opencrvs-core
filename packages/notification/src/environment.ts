/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * OpenCRVS is also distributed under the terms of the Civil Registration
 * & Healthcare Disclaimer located at http://opencrvs.org/license.
 *
 * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
 */
import { cleanEnv, str, port, url } from 'envalid'

export const env = cleanEnv(process.env, {
  HOST: str({ devDefault: 'localhost' }),
  PORT: port({ default: 2020 }),
  CERT_PUBLIC_KEY_PATH: str({ devDefault: '../../.secrets/public-key.pem' }),
  SENTRY_DSN: str({ default: undefined }),
  COUNTRY_CONFIG_URL: url({ devDefault: 'http://localhost:3040' }),
  USER_MANAGEMENT_URL: url({ devDefault: 'http://localhost:3030/' }),
  MONGO_URL: str({ devDefault: 'mongodb://localhost/notification' })
})
