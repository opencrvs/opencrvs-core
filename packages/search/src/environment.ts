/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * OpenCRVS is also distributed under the terms of the Civil Registration
 * & Healthcare Disclaimer located at http://opencrvs.org/license.
 *
 * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
 */

import { cleanEnv, str, port, url, num } from 'envalid'

export const env = cleanEnv(process.env, {
  HOST: str({ default: '0.0.0.0' }),
  PORT: port({ default: 9090 }),
  ES_HOST: str({ devDefault: 'localhost:9200' }),
  FHIR_URL: url({ devDefault: 'http://localhost:3447/fhir' }),
  CERT_PUBLIC_KEY_PATH: str({ default: '../../.secrets/public-key.pem' }),
  USER_MANAGEMENT_URL: url({ devDefault: 'http://localhost:3030/' }),
  MATCH_SCORE_THRESHOLD: num({ default: 1.0 }),
  OPENCRVS_INDEX_NAME: str({ default: 'ocrvs' }),
  APPLICATION_CONFIG_URL: url({ devDefault: 'http://localhost:2021/' }),
  HEARTH_MONGO_URL: str({ default: 'mongodb://localhost/hearth-dev' })
})
