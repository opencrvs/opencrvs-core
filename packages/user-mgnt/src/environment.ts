/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * OpenCRVS is also distributed under the terms of the Civil Registration
 * & Healthcare Disclaimer located at http://opencrvs.org/license.
 *
 * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
 */

import { cleanEnv, str, port, url, bool, num } from 'envalid'

export const env = cleanEnv(process.env, {
  HOST: str({ devDefault: 'localhost' }),
  PORT: port({ default: 3030 }),
  MONGO_URL: url({ devDefault: 'mongodb://localhost/user-mgnt' }),
  NOTIFICATION_SERVICE_URL: url({ devDefault: 'http://localhost:2020/' }),
  APPLICATION_CONFIG_URL: url({ devDefault: 'http://localhost:2021/' }),
  DOCUMENTS_URL: url({ devDefault: 'http://localhost:9050' }),
  COUNTRY_CONFIG_URL: url({ devDefault: 'http://localhost:3040/' }),
  CERT_PUBLIC_KEY_PATH: str({ devDefault: '../../.secrets/public-key.pem' }),
  SENTRY_DSN: str({ default: undefined }),
  QA_ENV: bool({ default: false }),
  RECORD_SEARCH_QUOTA: num({ default: 2000 }),

  FHIR_URL: url({ devDefault: 'http://localhost:3447/fhir' }),
  METRICS_URL: url({ devDefault: 'http://localhost:1050' })
})
