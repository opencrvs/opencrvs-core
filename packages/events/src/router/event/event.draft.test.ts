/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * OpenCRVS is also distributed under the terms of the Civil Registration
 * & Healthcare Disclaimer located at http://opencrvs.org/license.
 *
 * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
 */

import { orderBy } from 'lodash'
import {
  ActionStatus,
  ActionType,
  DraftInput,
  FullDocumentPath,
  generateUuid
} from '@opencrvs/commons'
import { mswServer } from '@events/tests/msw'
import {
  createTestClient,
  sanitizeForSnapshot,
  setupTestCase,
  UNSTABLE_EVENT_FIELDS
} from '@events/tests/utils'
import { env } from '@events/environment'

test('Throws error when creating a draft against non-existent id', async () => {
  const { user } = await setupTestCase()
  const client = createTestClient(user)

  const uuid = 'd09ca1db-98b6-4edc-891f-eebb33fea7a6'

  await expect(
    client.event.draft.create({
      eventId: uuid,
      type: ActionType.DECLARE,
      status: 'Accepted',
      transactionId: 'trnx-id'
    })
  ).rejects.toThrow(`Event not found with ID: ${uuid}`)
})

test('Throws error when creating a draft for event without assignment', async () => {
  const { users, generator } = await setupTestCase()
  const [defaultUser, anotherUser] = users
  const defaultClient = createTestClient(defaultUser)
  const anotherClient = createTestClient(anotherUser)

  const eventCreatedByOther = await anotherClient.event.create(
    generator.event.create()
  )

  await expect(
    defaultClient.event.draft.create({
      eventId: eventCreatedByOther.id,
      type: ActionType.DECLARE,
      status: 'Accepted',
      transactionId: 'trnx-id'
    })
  ).rejects.toThrow('You are not assigned to this event')
})

test('Throws error when creating a draft for invalid action', async () => {
  const { user, generator } = await setupTestCase()
  const client = createTestClient(user)

  const event = await client.event.create(generator.event.create())

  await expect(
    client.event.draft.create({
      eventId: event.id,
      type: ActionType.REGISTER,
      status: 'Accepted',
      transactionId: 'trnx-id'
    })
  ).rejects.toThrow(
    "Action 'REGISTER' cannot be performed on an event in 'CREATED' state. Available actions: READ, DECLARE, NOTIFY, DELETE"
  )
})

test('Allows creating draft for event without actions', async () => {
  const { user, generator } = await setupTestCase()
  const client = createTestClient(user)

  const event = await client.event.create(generator.event.create())

  const draftResponse = await client.event.draft.create({
    eventId: event.id,
    type: ActionType.DECLARE,
    status: 'Accepted',
    transactionId: 'trnx-id'
  })

  expect(
    sanitizeForSnapshot(draftResponse, UNSTABLE_EVENT_FIELDS)
  ).toMatchSnapshot()
})

test('Creating another draft replaces the previous one', async () => {
  const { user, generator } = await setupTestCase()
  const client = createTestClient(user)

  const event = await client.event.create(generator.event.create())

  await client.event.draft.create({
    eventId: event.id,
    type: ActionType.DECLARE,
    status: 'Accepted',
    transactionId: 'trnx-id-1',
    annotation: {
      comment: 'draft 1'
    }
  })

  const secondDraft = await client.event.draft.create({
    eventId: event.id,
    type: ActionType.DECLARE,
    status: 'Accepted',
    transactionId: 'trnx-id-2',
    annotation: {
      comment: 'draft 2'
    }
  })

  const drafts = await client.event.draft.list()

  expect(drafts).toHaveLength(1)
  expect(drafts[0]).toStrictEqual(secondDraft)

  expect(sanitizeForSnapshot(drafts, UNSTABLE_EVENT_FIELDS)).toMatchSnapshot()
})

test('Allows creating draft for event with actions', async () => {
  const { user, generator } = await setupTestCase()
  const client = createTestClient(user)

  const event = await client.event.create(generator.event.create())

  await client.event.actions.declare.request(
    generator.event.actions.declare(event.id, { keepAssignment: true })
  )

  const draftResponse = await client.event.draft.create({
    eventId: event.id,
    type: ActionType.VALIDATE,
    status: 'Accepted',
    transactionId: 'trnx-id'
  })

  expect(
    sanitizeForSnapshot(draftResponse, UNSTABLE_EVENT_FIELDS)
  ).toMatchSnapshot()
})

test('Creating a draft is idempotent', async () => {
  const { user, generator } = await setupTestCase()
  const client = createTestClient(user)

  const event = await client.event.create(generator.event.create())

  await client.event.actions.declare.request(
    generator.event.actions.declare(event.id, { keepAssignment: true })
  )

  const firstResponse = await client.event.draft.create({
    eventId: event.id,
    type: ActionType.VALIDATE,
    status: 'Accepted',
    transactionId: 'trnx-id',
    annotation: {
      comment: 'Should be visible'
    }
  })

  await client.event.draft.create({
    eventId: event.id,
    type: ActionType.VALIDATE,
    status: 'Accepted',
    transactionId: 'trnx-id',
    annotation: {
      comment: 'Should not be visible'
    }
  })

  const drafts = await client.event.draft.list()

  expect(drafts).toHaveLength(1)
  expect(drafts[0]).toEqual(firstResponse)
})

describe('Delete document references in drafts as side-effect', () => {
  const deleteUnreferencedDraftAttachmentsMock = vi.fn()
  const fileExistMock = vi.fn()

  function mockListener({
    request
  }: {
    response: Response
    request: Request
    requestId: string
  }) {
    if (!request.url.startsWith(`${env.DOCUMENTS_URL}/files`)) {
      return
    }

    if (request.method === 'DELETE') {
      deleteUnreferencedDraftAttachmentsMock(request.url, request.body)
    }

    if (request.method === 'HEAD') {
      fileExistMock(request.url, request.body)
    }
  }
  beforeEach(() => {
    mswServer.events.on('response:mocked', mockListener)
  })

  test('Delete previous draft attachments', async () => {
    const { user, generator, rng } = await setupTestCase(987)
    const client = createTestClient(user)
    const event = await client.event.create(generator.event.create())

    const getDraft = (n: number): DraftInput => {
      return {
        type: ActionType.DECLARE,
        declaration: {
          ...generator.event.actions.declare(event.id).declaration,
          'applicant.image': {
            type: 'image/png',
            originalFilename: `${n}-abcd.png`,
            path: `/ocrvs/${n}-4f095fc4-4312-4de2-aa38-86dcc0f71044.png` as FullDocumentPath
          }
        },
        transactionId: generateUuid(rng),
        eventId: event.id,
        status: ActionStatus.Requested
      }
    }

    // declaring 5 drafts with 4 different file attachments
    await client.event.draft.create(getDraft(1))
    await client.event.draft.create(getDraft(2))
    await client.event.draft.create(getDraft(3))
    await client.event.draft.create(getDraft(4))
    await client.event.draft.create(getDraft(5))

    // total 4 unreferenced draft attachments should be deleted
    expect(deleteUnreferencedDraftAttachmentsMock.mock.calls).toHaveLength(4)
    expect(deleteUnreferencedDraftAttachmentsMock.mock.calls).toMatchSnapshot()

    const drafts = await client.event.draft.list()

    expect(drafts).toHaveLength(1)
  })

  afterEach(() => {
    mswServer.events.removeListener('response:mocked', mockListener)
  })
})
