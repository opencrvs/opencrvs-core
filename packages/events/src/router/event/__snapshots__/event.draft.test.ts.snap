// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Allows creating draft for event with actions 1`] = `
{
  "action": {
    "annotation": null,
    "createdAt": "[sanitized]",
    "createdAtLocation": "[sanitized]",
    "createdBy": "[sanitized]",
    "createdByRole": "REGISTRATION_AGENT",
    "createdByUserType": "[sanitized]",
    "declaration": {},
    "status": "Accepted",
    "transactionId": "[sanitized]",
    "type": "VALIDATE",
  },
  "createdAt": "[sanitized]",
  "eventId": "[sanitized]",
  "id": "[sanitized]",
  "transactionId": "[sanitized]",
}
`;

exports[`Allows creating draft for event without actions 1`] = `
{
  "action": {
    "annotation": null,
    "createdAt": "[sanitized]",
    "createdAtLocation": "[sanitized]",
    "createdBy": "[sanitized]",
    "createdByRole": "REGISTRATION_AGENT",
    "createdByUserType": "[sanitized]",
    "declaration": {},
    "status": "Accepted",
    "transactionId": "[sanitized]",
    "type": "DECLARE",
  },
  "createdAt": "[sanitized]",
  "eventId": "[sanitized]",
  "id": "[sanitized]",
  "transactionId": "[sanitized]",
}
`;

exports[`Creating another draft replaces the previous one 1`] = `
[
  {
    "action": {
      "annotation": {
        "comment": "draft 2",
      },
      "createdAt": "[sanitized]",
      "createdAtLocation": "[sanitized]",
      "createdBy": "[sanitized]",
      "createdByRole": "REGISTRATION_AGENT",
      "createdByUserType": "[sanitized]",
      "declaration": {},
      "status": "Accepted",
      "transactionId": "[sanitized]",
      "type": "DECLARE",
    },
    "createdAt": "[sanitized]",
    "eventId": "[sanitized]",
    "id": "[sanitized]",
    "transactionId": "[sanitized]",
  },
]
`;

exports[`Delete document references in drafts as side-effect > Delete previous draft attachments 1`] = `
[
  [
    "http://localhost:9050/files/ocrvs/1-4f095fc4-4312-4de2-aa38-86dcc0f71044.png",
    ReadableStream {
      "constructor": [Function],
      Symbol(kType): "ReadableStream",
      Symbol(kState): {
        "controller": ReadableStreamDefaultController {
          Symbol(kType): "ReadableStreamDefaultController",
          Symbol(kState): {
            "cancelAlgorithm": undefined,
            "closeRequested": true,
            "highWaterMark": 1,
            "pullAgain": false,
            "pullAlgorithm": undefined,
            "pulling": false,
            "queue": [],
            "queueTotalSize": 0,
            "sizeAlgorithm": undefined,
            "started": true,
            "stream": [Circular],
          },
        },
        "disturbed": false,
        "reader": undefined,
        "state": "closed",
        "storedError": undefined,
        "transfer": {
          "port1": undefined,
          "port2": undefined,
          "promise": undefined,
          "writable": undefined,
        },
      },
      Symbol(nodejs.webstream.isClosedPromise): {
        "promise": Promise {},
        "reject": [Function],
        "resolve": [Function],
      },
      Symbol(nodejs.webstream.controllerErrorFunction): [Function],
    },
  ],
  [
    "http://localhost:9050/files/ocrvs/2-4f095fc4-4312-4de2-aa38-86dcc0f71044.png",
    ReadableStream {
      "constructor": [Function],
      Symbol(kType): "ReadableStream",
      Symbol(kState): {
        "controller": ReadableStreamDefaultController {
          Symbol(kType): "ReadableStreamDefaultController",
          Symbol(kState): {
            "cancelAlgorithm": undefined,
            "closeRequested": true,
            "highWaterMark": 1,
            "pullAgain": false,
            "pullAlgorithm": undefined,
            "pulling": false,
            "queue": [],
            "queueTotalSize": 0,
            "sizeAlgorithm": undefined,
            "started": true,
            "stream": [Circular],
          },
        },
        "disturbed": false,
        "reader": undefined,
        "state": "closed",
        "storedError": undefined,
        "transfer": {
          "port1": undefined,
          "port2": undefined,
          "promise": undefined,
          "writable": undefined,
        },
      },
      Symbol(nodejs.webstream.isClosedPromise): {
        "promise": Promise {},
        "reject": [Function],
        "resolve": [Function],
      },
      Symbol(nodejs.webstream.controllerErrorFunction): [Function],
    },
  ],
  [
    "http://localhost:9050/files/ocrvs/3-4f095fc4-4312-4de2-aa38-86dcc0f71044.png",
    ReadableStream {
      "constructor": [Function],
      Symbol(kType): "ReadableStream",
      Symbol(kState): {
        "controller": ReadableStreamDefaultController {
          Symbol(kType): "ReadableStreamDefaultController",
          Symbol(kState): {
            "cancelAlgorithm": undefined,
            "closeRequested": true,
            "highWaterMark": 1,
            "pullAgain": false,
            "pullAlgorithm": undefined,
            "pulling": false,
            "queue": [],
            "queueTotalSize": 0,
            "sizeAlgorithm": undefined,
            "started": true,
            "stream": [Circular],
          },
        },
        "disturbed": false,
        "reader": undefined,
        "state": "closed",
        "storedError": undefined,
        "transfer": {
          "port1": undefined,
          "port2": undefined,
          "promise": undefined,
          "writable": undefined,
        },
      },
      Symbol(nodejs.webstream.isClosedPromise): {
        "promise": Promise {},
        "reject": [Function],
        "resolve": [Function],
      },
      Symbol(nodejs.webstream.controllerErrorFunction): [Function],
    },
  ],
  [
    "http://localhost:9050/files/ocrvs/4-4f095fc4-4312-4de2-aa38-86dcc0f71044.png",
    ReadableStream {
      "constructor": [Function],
      Symbol(kType): "ReadableStream",
      Symbol(kState): {
        "controller": ReadableStreamDefaultController {
          Symbol(kType): "ReadableStreamDefaultController",
          Symbol(kState): {
            "cancelAlgorithm": [Function],
            "closeRequested": false,
            "highWaterMark": 1,
            "pullAgain": false,
            "pullAlgorithm": [Function],
            "pulling": false,
            "queue": [],
            "queueTotalSize": 0,
            "sizeAlgorithm": [Function],
            "started": true,
            "stream": [Circular],
          },
        },
        "disturbed": false,
        "reader": undefined,
        "state": "readable",
        "storedError": undefined,
        "transfer": {
          "port1": undefined,
          "port2": undefined,
          "promise": undefined,
          "writable": undefined,
        },
      },
      Symbol(nodejs.webstream.isClosedPromise): {
        "promise": Promise {},
        "reject": [Function],
        "resolve": [Function],
      },
      Symbol(nodejs.webstream.controllerErrorFunction): [Function],
    },
  ],
]
`;
